File: app/src/main/java/co/early/password123/ui/passwordchooser/TextWatcherCheckPassword.java
package co.early.password123.ui.passwordchooser;

import android.text.Editable;
import android.text.TextWatcher;

import co.early.asaf.core.Affirm;
import co.early.pwned.Pwned;



public class TextWatcherCheckPassword implements TextWatcher {

    private final Pwned pwned;

    public TextWatcherCheckPassword(Pwned pwned) {
        this.pwned = Affirm.notNull(pwned);
    }

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count, int after) {

    }

    @Override
    public void onTextChanged(CharSequence s, int start, int before, int count) {

    }

    @Override
    public void afterTextChanged(Editable s) {
        pwned.setPasswordToCheck(s.toString());
    }

}

File: app/src/main/java/co/early/password123/ui/passwordchooser/TextWatcherCheckPassword.kt
package co.early.password123.ui.passwordchooser

import android.text.Editable
import android.text.TextWatcher

import co.early.asaf.core.Affirm
import co.early.pwned.Pwned


class TextWatcherCheckPassword(pwned: Pwned) : TextWatcher {

    private val pwned: Pwned

    init {
        this.pwned = Affirm.notNull(pwned)
    }

    override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {

    }

    override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {

    }

    override fun afterTextChanged(s: Editable) {
        pwned.setPasswordToCheck(s.toString())
    }

}

